lib/task_005.rb

```diff
@@ -0,0 +1,124 @@
+require_relative 'read_file'
+
+# They tend to form in lines; the submarine helpfully produces a list of nearby lines of vents (your puzzle input)
+# for you to review. For example:
+#
+# 0,9 -> 5,9
+# 8,0 -> 0,8
+# 9,4 -> 3,4
+# 2,2 -> 2,1
+# 7,0 -> 7,4
+# 6,4 -> 2,0
+# 0,9 -> 2,9
+# 3,4 -> 1,4
+# 0,0 -> 8,8
+# 5,5 -> 8,2
+#
+# Each line of vents is given as a line segment in the format x1,y1 -> x2,y2 where x1,y1 are the coordinates of
+# one end the line segment and x2,y2 are the coordinates of the other end. These line segments include the points
+# at both ends. In other words:
+#
+#     An entry like 1,1 -> 1,3 covers points 1,1, 1,2, and 1,3.
+#     An entry like 9,7 -> 7,7 covers points 9,7, 8,7, and 7,7.
+#
+# For now, only consider horizontal and vertical lines: lines where either x1 = x2 or y1 = y2.
+#
+# So, the horizontal and vertical lines from the above list would produce the following diagram:
+#
+# .......1..
+# ..1....1..
+# ..1....1..
+# .......1..
+# .112111211
+# ..........
+# ..........
+# ..........
+# ..........
+# 222111....
+#
+# In this diagram, the top left corner is 0,0 and the bottom right corner is 9,9. Each position is shown as the
+# number of lines which cover that point or . if no line covers that point. The top-left pair of 1s, for example,
+# comes from 2,2 -> 2,1; the very bottom row is formed by the overlapping lines 0,9 -> 5,9 and 0,9 -> 2,9.
+#
+# To avoid the most dangerous areas, you need to determine the number of points where at least two lines overlap.
+#  In the above example, this is anywhere in the diagram with a 2 or larger - a total of 5 points.
+#
+# Consider only horizontal and vertical lines. At how many points do at least two lines overlap?
+
+def parse_input(input_data)
+  input_data.split("\n").map { |line| parse_line(line) }
+end
+
+def parse_line(line)
+  point1, point2 = line.split(" -> ")
+
+  {
+    point1: parse_point(point1),
+    point2: parse_point(point2)
+  }
+end
+
+def parse_point(point)
+  point.split(',').map { |value| value.to_i }
+end
+
+def create_diagram(lines)
+  x = lines.max_by { |line| line[:point1][0] || line[:point2][0] }
+  y = lines.max_by { |line| line[:point1][1] || line[:point2][1] }
```

Comment:
I'm not sure what is the intention behind those one:
```ruby
line[:point1][0] || line[:point2][0]
line[:point1][1] || line[:point2][1]
```

Bur wouldn't it always return the first value (meaning `line[:point1][0]` and `line[:point1][1]`)?

Reply:
The line = { point1: [x, y], point2[x, y] }, so I need to find line with max of x, x of point1 or point2, so thats why : 

> line[:point1][0] || line[:point2][0]

next, the same with y 
I need to know this to find size of diagram, diagram need to cover every points(x, y)

Reply:
```ruby
x = lines.max_by { |line| line[:point1][0] || line[:point2][0] }
```

I believe the above would fine the line with max of `line[:point1][0]` and `line[:point2][0] ` will be ignored.

Since `line[:point1][0]` is not `nil`, the `line[:point1][0] || line[:point2][0]` expression would always return `line[:point1][0]`.

In the other words, assuming `line[:point1][0]` would not be `nil`, then:

```ruby
x = lines.max_by { |line| line[:point1][0] || line[:point2][0] }
```

would be the same as

```ruby
x = lines.max_by { |line| line[:point1][0] }
```

Reply:
I guess you may need something lines this:

```ruby
x = lines.max_by { |line| [line[:point1][0], line[:point2][0]].max }
```

Reply:
Yes, I see now, becouse  || operator give us first true value.